src/source-map-tree.ts

```diff
@@ -115,22 +115,35 @@ export default class SourceMapTree {
     if (line >= mappings.length) return null;
 
     const segments = mappings[line];
-    const index = binarySearch(segments, column, segmentComparator);
 
-    // If we can't find an segment that lines up to this column, then we can't
-    // trace it further.
-    if (index < 0) return null;
+    if (segments.length === 0) return null;
+
+    let index = binarySearch(segments, column, segmentComparator);
+
+    // If we can't find an segment that lines up to this column, we use the
+    // closest segment before it, and offset the source column
+    if (index < 0) {
+      index = ~index - 1;
+      if (index < 0) return null; // we come before any mapped segment
+    }
+
     const segment = segments[index];
+    if (!segment) return null;
 
     // 1-length segments only move the current generated column, there's no
     // source information to gather from it.
     if (segment.length === 1) return null;
     const source = this.sources[segment[1]];
 
+    // calculate the new column based on the difference between the column we
+    // were searching for, and the start of the segment we found
+    const offset = column - segment[0];
+    const originalCol = segment[3] + offset;
```

Comment:
I think this requires a check that the new column is still part of this segment. Eg, if we're offseting 10 chars but the segment itself only reaches 4 chars.

Reply:
I am not sure I follow 100%. Can you think of a mapping that would cause this?,

Reply:
I think I worked out what you mean. It was my misunderstanding.
I have update the code to behave like mozilla's source-map where a non exact match will return the closest lowest segments source location